includes/data-stores/class-wcs-orders-table-subscription-data-store.php

```diff
@@ -460,6 +472,46 @@ public function update_order_meta( &$subscription ) {
 		parent::update_order_meta( $subscription );
 	}
 
+	/**
+	 * Undocumented function
+	 *
+	 * @param WC_Subscription $subscription The subscription object to save updates for.
+	 * @param bool            $backfill     Whether to backfill the subscription's meta data.
+	 */
+	protected function persist_updates( &$subscription, $backfill = true ) {
```

Comment:
In our datastore we had 1 function for saving meta data when a subscription is saved `update_order_meta()`. That function works by taking our internal props (billing interval, period, dates, and other data that isn't meta) and storing it in meta before the subscription is saved. WC than would take those meta values and save them later in the flow. 

See [`OrdersTableDataStore::persist_save()`](https://github.com/woocommerce/woocommerce/blob/7cd1a030430d7215b4548d6a7037f79abd0dcb72/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php#L1983-L2002)

However `persist_save()` is only called when an order is created ([search ref](https://github.com/woocommerce/woocommerce/search?q=%24this-%3Epersist_save%28)). Not when it is updated. 

Therefore in order for us to save our internal meta properties on update (not create) we need to use a new function. 

This is where `persist_updates()` comes in. There is no equivalent to `update_order_meta()` that is called during updating. So while the functions are somewhat similar, they have to use slightly different methods to save the data which is unfortunate to say the least. 

I'll take a look again on monday to see if there is another way around this.